xarray/core/groupby.py

```diff
@@ -404,6 +418,8 @@ def __init__(
         self._groups = None
         self._dims = None
 
+    # TODO: is this correct? Should we be returning the dims of the result? This
+    # will use the original dim where we're grouping by a coord.
```

Comment:
Is the existing code for this property correct? Currently `x.groupby(foo).dims` != `x.groupby(foo).sum(...).dims` when we're grouping by an non-indexed coord

Reply:
It returns the dims for the first group, so you can decide what `dim` values can be passed to `.mean`. But I agree it is confusing. Maybe we should deprecate and remove. This use-case is also served by implementing `GroupBy.__getitem__`